### PR TITLE
Update Groovy and Kotlin version ranges in :updateInitPluginTemplateVersionFile

### PR DESCRIPTION
**Pull request by eskatos**
respectively to `2.5.+` and `1.3.+`.

### CODE REVIEW

File: subprojects/build-init/build-init.gradle.kts [hunk 1]
```diff
@@ -59,7 +59,7 @@ tasks {
 
             findLatest("scalatest", "org.scalatest:scalatest_${versionProperties["scala"]}:(3.0,)", versionProperties)
             findLatest("scala-xml", "org.scala-lang.modules:scala-xml_${versionProperties["scala"]}:latest.release", versionProperties)
-            findLatest("groovy", "org.codehaus.groovy:groovy:(2.4,2.5]", versionProperties)
+            findLatest("groovy", "org.codehaus.groovy:groovy:(2.5,2.6]", versionProperties)
```
Comment (wolfs) [inline subprojects/build-init/build-init.gradle.kts]: ❌ The spock version below still needs to be updated (to `1.2` I guess). Can we make the spock version update automatic as well?

Comment (wolfs) [inline subprojects/build-init/build-init.gradle.kts]: I don't quite understand the version ranges. Doesn't the range mean that `2.6` would be included and `2.5` excluded? I think it should be the other way round, right?

File: subprojects/build-init/build-init.gradle.kts [hunk 2]
```diff
@@ -59,7 +59,7 @@ tasks {
 
             findLatest("scalatest", "org.scalatest:scalatest_${versionProperties["scala"]}:(3.0,)", versionProperties)
```
Comment (wolfs) [inline subprojects/build-init/build-init.gradle.kts]: Can we now use a Scala version bigger than `2.12.x`?

File: subprojects/build-init/build-init.gradle.kts (hunk 1 above)
Comment (eskatos) [reply subprojects/build-init/build-init.gradle.kts]: Right. Seems they were wrong? before.
It's good that got documented recently :) https://github.com/gradle/gradle/commit/44630395e54138108e98cd69ccc3a355b0817a24#diff-60edd0ca069e1e63e58f23bc7df3878bR35

File: subprojects/build-init/build-init.gradle.kts (hunk 2 above)
Comment (eskatos) [reply subprojects/build-init/build-init.gradle.kts]: I'll leave that out of this PR's scope

**Pull request merged**